src/ZEO/asyncio/client.py

```diff
@@ -199,9 +205,12 @@ def finish_connect(self, protocol_version):
 
     exception_type_type = type(Exception)
     def message_received(self, data):
-        msgid, async, name, args = decode(data)
+        msgid, async, name, args = self.decode(data)
         if name == '.reply':
             future = self.futures.pop(msgid)
+            if isinstance(future, tuple):
+                future = self.futures.pop(future)
```

Comment:
Would it be possible to back this part of it out for now so that when it is needed it goes with a more logical PR?

I understand if you'd rather not back it out, and I won't complain. I just had to ask :)